### PR TITLE
Initial module configuration

### PR DESCRIPTION
**Pull request by atsikham**
### Description
This module is based on previous work: azure-bi [module](https://github.com/epiphany-platform/m-azure-basic-infrastructure), azure-k8s [module](https://github.com/epiphany-platform/m-azure-kubernetes-service), aws-bi [module](https://github.com/epiphany-platform/m-aws-basic-infrastructure) and [spike](https://github.com/epiphany-platform/epiphany/issues/1442) about eks configuration.

### What is different from spike
1. AWS _account_id_ is not passed as a [variable](https://github.com/epiphany-platform/epiphany/blob/40b3373090e937f741aabb2b7712a136f7e6a6fe/docs/design-docs/eks/terraform-scripts/cluster-autoscaler.tf#L47), it's not necessary, instead it's retrieved from datasource as in [example](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/examples/irsa/outputs.tf).
2. Used _private_ subnets for EKS worker nodes.
3. As in the spike terraform _12.x_ was used and for modules we are going to start with _13.x_, some versions are changed, for example `iam_assumable_role_admin` module version.

### Problems found (can be addressed as separate tasks/spikes)
1. I was not able to install 1.18 k8s version with unclear error. All [examples](https://github.com/terraform-aws-modules/terraform-aws-eks/search?q=cluster_version) with this eks terraform module use 1.17 version, when 1.19 has been recently released, so I don't think it's maintained well. We could consider using embedded terraform [resource](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_cluster) or another tool, [pulumi ](https://aws.amazon.com/ru/blogs/apn/how-to-easily-deploy-an-amazon-eks-cluster-with-pulumi/) for example. But definitely it have to be stable and well supported.
2. There is a [bug](https://github.com/terraform-aws-modules/terraform-aws-eks/issues/568), check comments there, and eks destroy finishes with non-zero error code, but removes all the resources. So for destroy we need to use `make -k destroy`.
3. Seems that we use [deprecated](https://artifacthub.io/packages/helm/helm-stable/cluster-autoscaler) helm chart for cluster autoscaler. This one is [new](https://github.com/kubernetes/autoscaler). Probably @mum0l can clarify.
4. Infrastructure part could be moved to aws-bi module.

### CODE REVIEW

**Review by mkyc — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ARG ARG_HOST_GID=1000
+RUN chown -R $ARG_HOST_UID:$ARG_HOST_GID \
+    $M_WORKDIR \
+    $M_RESOURCES
+
+USER $ARG_HOST_UID:$ARG_HOST_GID
+# Set HOME to directory with necessary permissions for current user
+ENV HOME=$M_WORKDIR
+
+RUN helm repo add stable https://kubernetes-charts.storage.googleapis.com
```
[the remark this answers is not in the document]
Comment (atsikham) [reply Dockerfile:45]: tried do this and finally was able to fix issues only with current user usage

File: resources/defaults.mk [hunk 2]
```diff
@@ -0,0 +1,10 @@
+M_NAME ?= epiphany
+M_VPC_ID ?= unset
+M_PUBLIC_SUBNET_ID ?= unset
+M_K8S_VERSION ?= 1.17
+M_K8S_AUTOSCALER_VERSION ?= v1.17.3
```
[the remark this answers is not in the document]
Comment (atsikham) [reply resources/defaults.mk:5]: @mkyc this is a question to discuss. I think so, there is extensive description in this PR.

File: resources/terraform/main.tf [hunk 3]
```diff
@@ -0,0 +1,104 @@
+data "aws_vpc" "vpc" {
+  id = var.vpc_id
+}
+
+# Need to create private subnets as it's not done by awsbi module.
```
[the remark this answers is not in the document]
Comment (atsikham) [reply resources/terraform/main.tf:5]: @mkyc this is a question to discuss. I think so, there is extensive description in this PR.

File: resources/terraform/modules/awsks/versions.tf [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      version = "~> 2.1"
+    }
+
+    random = {
+      version = "~> 2.1"
+    }
+
+    template = {
+      version = "~> 2.1"
+    }
```
[the remark this answers is not in the document]
Comment (atsikham) [reply resources/terraform/modules/awsks/versions.tf:32]: This is present in [spike](https://github.com/epiphany-platform/epiphany/blob/feature/eks-cluster-creation/docs/design-docs/eks/terraform-scripts/main.tf) and [example](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/examples/irsa/main.tf) of module usage and module [documentation](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/README.md#requirements). 
Do we need to check? I wonder if we need to do such things in this PR as it wasn't aim of the task.

File: resources/terraform/output.tf [hunk 5]
```diff
@@ -0,0 +1,9 @@
+output "kubeconfig" {
+  description = "Kubeconfig as generated by the module"
+  value       = module.awsks.kubeconfig
+  sensitive   = true
+}
+
+output "vpc" {
```
[the remark this answers is not in the document]
Comment (atsikham) [reply resources/terraform/output.tf]: it was used for debug, removed

File: resources/terraform/modules/awsks/versions.tf (hunk 4 above)
[the remark this answers is not in the document]
Comment (mkyc) [reply resources/terraform/modules/awsks/versions.tf]: Ok, so more general question would be if we want to use this eks module at all? Also from this PR description I assume there should be new task about changing it to pure terraform aws provider implementation.

File: resources/terraform/main.tf (hunk 3 above)
[the remark this answers is not in the document]
Comment (mkyc) [reply resources/terraform/main.tf:5]: Yeah, so we should probably create those.

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkyc) [reply Dockerfile:45]: Ok, let's not focus on that one for now.

**Comment by atsikham** [comment]
Created new tasks: #6, #7, and [infrastructure](https://github.com/epiphany-platform/m-aws-basic-infrastructure/issues/6) related.

File: resources/terraform/modules/awsks/autoscaler.tf [hunk 6]
```diff
@@ -0,0 +1,51 @@
+data "aws_caller_identity" "current" {}
+
+resource "helm_release" "cluster-autoscaler" {
```
[the remark this answers is not in the document]
Comment (atsikham) [reply resources/terraform/modules/awsks/autoscaler.tf:3]: Added this to #7

File: resources/terraform/modules/awsks/autoscaler-iam.tf [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    sid       = "clusterAutoscalerOwn"
+    effect    = "Allow"
+
+    actions   = [
+      "autoscaling:SetDesiredCapacity",
+      "autoscaling:TerminateInstanceInAutoScalingGroup",
+      "autoscaling:UpdateAutoScalingGroup",
+    ]
+
+    resources = ["*"]
```
[the remark this answers is not in the document]
Comment (atsikham) [reply resources/terraform/modules/awsks/autoscaler-iam.tf:49]: No, I'm not sure. It was done so in the spike and module [examples](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/examples/irsa/irsa.tf). I'll create #8 task soon.

File: resources/terraform/modules/awsks/autoscaler-iam.tf [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+    actions   = [
+      "autoscaling:DescribeAutoScalingGroups",
+      "autoscaling:DescribeAutoScalingInstances",
+      "autoscaling:DescribeLaunchConfigurations",
+      "autoscaling:DescribeTags",
+      "ec2:DescribeLaunchTemplateVersions",
+    ]
+
+    resources = ["*"]
```
[the remark this answers is not in the document]
Comment (atsikham) [reply resources/terraform/modules/awsks/autoscaler-iam.tf:36]: The same as in next comment.

**Review by mkyc — APPROVED** [review]
LGTM

**Review by seriva — APPROVED** [review]

**Review by pyrkamarcin — APPROVED** [review]

**Pull request merged**